tiledb/sm/cpp_api/query.h

```diff
@@ -187,6 +183,12 @@ class Query {
     return query_;
   }
 
+  /** Returns a shared pointer to the C TileDB subarray object associated with
```

Comment:
remove, expose a getter to the C++ subarray instead, allowing the caller to fetch the c-ptr from there.